### DISCUSSION TITLE
Add Vue component to nav bar

### DISCUSSION THREAD

**Discussion by ljcremer** [vuepress/core#383] [Q&A] [answered]
Is it possible to add a vue component into the nav bar?

We would like to add our own search component into the navbar instead of the other search options.

**Comment by meteorlxy** [comment, marked as answer]
https://v2.vuepress.vuejs.org/advanced/cookbook/extending-a-theme.html#extend-default-theme

**Reply by ljcremer** [reply]
Thanks for the guidance. I am still not sure how to do it exactly. Should I change the layout component or the navbar component?

Or is it maybe just possible to replace the NavbarSearch component only with my own component?

**Reply by Mister-Hope** [reply]
> Thanks for the guidance. I am still not sure how to do it exactly. Should I change the layout component or the navbar component?
You should try overiding Navbar component

> Or is it maybe just possible to replace the NavbarSearch component only with my own component?

Yes it's possible, you should try to unregister the original Search component and register with yours.

**Reply by xiaoyunwu** [reply]
how do I register a component? :-(

**Reply by Mister-Hope** [reply]
use client app enhance file

**Reply by xiaoyunwu** [reply]
my question should be unregister instead. :-(

**Comment by xiaoyunwu** [comment]
thanks, link to some example?
